packages/framework-provider-local/src/library/read-model-adapter.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  BoosterConfig,
+  Filter,
+  Logger,
+  ReadModelEnvelope,
+  ReadModelInterface,
+  UUID,
+} from '@boostercloud/framework-types'
+import { ReadModelRegistry } from '../services/read-model-registry'
+
+export async function rawReadModelEventsToEnvelopes(
+  config: BoosterConfig,
+  logger: Logger,
+  rawEvents: Array<unknown>
+): Promise<Array<ReadModelEnvelope>> {
+  return rawEvents.map((event) => event as ReadModelEnvelope)
+}
+
+export async function fetchReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  readModelID: UUID
+): Promise<ReadModelInterface> {
+  const response = await db.query({ typeName: readModelName, value: { id: readModelID } })
+  const item = response[0]
+  if (!item) {
+    console.log(`[ReadModelAdapter#fetchReadModel] Failed to fetch read model ${readModelName} with ID ${readModelID}`)
+  } else {
+    logger.debug(
+      `[ReadModelAdapter#fetchReadModel] Loaded read model ${readModelName} with ID ${readModelID} with result:`,
+      item.value
+    )
+  }
+  return item?.value
+}
+
+export async function storeReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  readModel: ReadModelInterface
+): Promise<void> {
+  await db.store({ typeName: readModelName, value: readModel })
+  logger.debug('[ReadModelAdapter#storeReadModel] Read model stored')
+}
+
+export async function searchReadModel(
+  db: ReadModelRegistry,
+  _config: BoosterConfig,
+  logger: Logger,
+  readModelName: string,
+  filters: Record<string, Filter<unknown>>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+): Promise<Array<any>> {
```

Comment:
```suggestion
): Promise<Array<ReadModelInterface>> {
```

Reply:
This will fail with the following error message:
```text
@boostercloud/framework-provider-local: src/index.ts(38,5): error TS2322: Type '(_config: BoosterConfig, logger: Logger, readModelName: string, filters: Record<string, Filter<unknown>>) => Promise<ReadModelInterface[]>' is not assignable to type '<TReadModel extends ReadModelInterface>(config: BoosterConfig, logger: Logger, entityTypeName: string, filters: Record<string, Filter<unknown>>) => Promise<...>'.
@boostercloud/framework-provider-local:   Type 'Promise<ReadModelInterface[]>' is not assignable to type 'Promise<TReadModel[]>'.
@boostercloud/framework-provider-local:     Type 'ReadModelInterface[]' is not assignable to type 'TReadModel[]'.
@boostercloud/framework-provider-local:       Type 'ReadModelInterface' is not assignable to type 'TReadModel'.
@boostercloud/framework-provider-local:         'ReadModelInterface' is assignable to the constraint of type 'TReadModel', but 'TReadModel' could be instantiated with a different subtype of constraint 'ReadModelInterface'.
```

I spent like 1 hour trying to figure it out, but ended up leaving an `any`

Reply:
WTF! Ok then